### PR TITLE
feat: add cpp language detector

### LINKED ISSUE #61: Create a language detector for C++ (~100 new lines of code)
We'd like to measure the quality of apps written in C++.

There are "language detectors" to determine what language is on which path.

It's necessary to implement **CppLanguageDetector** the same way as is [JavaScriptLanguageDetector](https://github.com/DXHeroes/dx-scanner/blob/master/src/detectors/JavaScript/JavaScriptLanguageDetector.ts) implemented. It's around 100 new lines of code including tests.

More info in [our Contribution guide](https://github.com/DXHeroes/dx-scanner/blob/master/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by Sehsyha**
Fixes #61

### CODE REVIEW

File: src/detectors/Cpp/CppLanguageDetector.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export class CppLanguageDetector implements ILanguageDetector {
+  private fileInspector: IFileInspector;
+  constructor(@inject(Types.IFileInspector) fileInspector: IFileInspector) {
+    this.fileInspector = fileInspector;
+  }
+
+  async detectLanguage(): Promise<LanguageAtPath[]> {
+    const result: LanguageAtPath[] = [];
+
+    const cppFiles = await this.fileInspector.scanFor(fileExtensionRegExp(['cpp', 'hpp']), '/');
```
Comment (prokopsimek) [inline src/detectors/Cpp/CppLanguageDetector.ts]: The same as https://github.com/DXHeroes/dx-scanner/pull/90#discussion_r336892921

Comment (Sehsyha) [reply src/detectors/Cpp/CppLanguageDetector.ts]: Do you know which file can I search for ? I don't know if there's a famous package manager or something like this.

Comment (prokopsimek) [reply src/detectors/Cpp/CppLanguageDetector.ts]: @Sehsyha I have found this: https://stackoverflow.com/questions/27866965/does-c-have-a-package-manager-like-npm-pip-gem-etc My colleague told me, there is no unified definition file. So leave it as is. :) thanks

**Pull request merged**